sdk/storage/storage-file-share/test/fileclient.spec.ts

```diff
@@ -1576,3 +1577,1014 @@ describe("ShareFileClient - Verify Name Properties", () => {
+    assert.deepStrictEqual(result.ranges![0], { start: 512, end: 1535 });
+  });
+
+  it("getRangeListDiff with share snapshot", async function (this: Context) {
+    await fileClient.create(512 * 4 + 1);
+    await fileClient.uploadRange("Hello", 0, 5);
+
+    const snapshotRes = await shareClient.createSnapshot();
+    assert.ok(snapshotRes.snapshot);
+
+    await fileClient.clearRange(0, 1024);
+    await fileClient.uploadRange("World", 1023, 5);
+
+    const snapshotRes2 = await shareClient.createSnapshot();
+    assert.ok(snapshotRes2.snapshot);
+
+    await fileClient.uploadRange("Hello", 0, 5);
+
+    const fileClientWithShareSnapShot = fileClient.withShareSnapshot(snapshotRes2.snapshot!);
+    const result = await fileClientWithShareSnapShot.getRangeListDiff(snapshotRes.snapshot!);
+
+    assert.ok(result.clearRanges);
+    assert.deepStrictEqual(result.clearRanges!.length, 1);
+    assert.deepStrictEqual(result.clearRanges![0], { start: 0, end: 511 });
+
+    assert.ok(result.ranges);
+    assert.deepStrictEqual(result.ranges!.length, 1);
+    assert.deepStrictEqual(result.ranges![0], { start: 512, end: 1535 });
+  });
+
+  it("download", async () => {
+    await fileClient.create(content.length);
+    await fileClient.uploadRange(content, 0, content.length);
+    const result = await fileClient.download();
+    assert.deepStrictEqual(await bodyToString(result, content.length), content);
+  });
+
+  it("listHandles should work", async () => {
+    await fileClient.create(10);
+
+    const result = (await fileClient.listHandles().byPage().next()).value;
+    if (result.handleList !== undefined && result.handleList.length > 0) {
+      const handle = result.handleList[0];
+      assert.notDeepEqual(handle.handleId, undefined);
+      assert.notDeepEqual(handle.path, undefined);
+      assert.notDeepEqual(handle.fileId, undefined);
+      assert.notDeepEqual(handle.sessionId, undefined);
+      assert.notDeepEqual(handle.clientIp, undefined);
+      assert.notDeepEqual(handle.openTime, undefined);
+    }
+  });
+
+  it("forceCloseAllHandles should work", async () => {
+    await fileClient.create(10);
+
+    assert.deepStrictEqual(
+      await fileClient.forceCloseAllHandles(),
+      { closedHandlesCount: 0, closeFailureCount: 0 },
+      "Error in forceCloseAllHandles"
+    );
+  });
+
+  it("forceCloseHandle should work", async () => {
+    await fileClient.create(10);
+
+    const result = (await fileClient.listHandles().byPage().next()).value;
+    if (result.handleList !== undefined && result.handleList.length > 0) {
+      const handle = result.handleList[0];
+      await dirClient.forceCloseHandle(handle.handleId);
+    }
+  });
+
+  it("rename", async () => {
+    await fileClient.create(1024);
+    const destFileName = recorder.getUniqueName("destfile");
+    const destFileNameWithTrailingDots = destFileName + "....";
+    const result = await fileClient.rename(destFileNameWithTrailingDots);
```

Comment:
should we check allowsourcetrailingdots setting is same in the dest file client.